pkg/util/tags_builder.go

```diff
@@ -105,6 +120,14 @@ func (tb *TagsBuilder) Less(i, j int) bool {
 	return tb.data[i] < tb.data[j]
 }
 
+// Slice returns a shared slice of tb's internal data.
+func (tb *TagsBuilder) Slice(i, j int) *TagsBuilder {
+	return &TagsBuilder{
+		data: tb.data[i:j],
+		hash: tb.hash[i:j],
```

Comment:
This looks pretty scary -- that slice will share the same underlying array, so any `slice.Append(..)` will re-use storage slots that the parent TagsBuilder is also using.

```go
tb1 := NewTagsBuilder(..) // with 15 items
tb2 := tb1.Slice(0, 10)
tb2.Append("foo")
assert.Equal(t, "foo", tb1.Get()[10])
```
Maybe it would be better for all tagsets to be immutable?  Or to have an immutability flag that is set to true in the TagsBuilder returned from Slice?